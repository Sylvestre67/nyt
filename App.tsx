import React, { Component } from "react";
import { Platform, StyleSheet, Text, View } from "react-native";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import createSagaMiddleware from "redux-saga";

import appStore from "./store/reducers";
import rootSaga from "./store/sagas";

import HomeScreen from "./screens/HomeScreen";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(appStore, applyMiddleware(sagaMiddleware));

// then run the saga
sagaMiddleware.run(rootSaga);

interface Props {}
export default class App extends Component<Props> {
  render() {
    return (
      <Provider store={store}>
        <HomeScreen />
      </Provider>
    );
  }
}
