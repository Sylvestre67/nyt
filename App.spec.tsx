import React from "react";
import { mount, ReactWrapper } from "enzyme";

import App from "./App";

let component: ReactWrapper;

describe("app", () => {
  beforeAll(() => {
    component = mount(<App />);
  });

  it("should have a test", () => {
    expect(component).toMatchSnapshot();
  });
});
