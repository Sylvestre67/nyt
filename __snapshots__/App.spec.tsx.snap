// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should have a test 1`] = `
<App>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(HomeScreen)>
      <HomeScreen
        articles={
          Object {
            "hasError": false,
            "isLoading": false,
            "list": null,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5FCFF",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Welcome
              </Text>
            </Text>
            <Text
              style={
                Object {
                  "color": "#333333",
                  "marginBottom": 5,
                  "textAlign": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333",
                    "marginBottom": 5,
                    "textAlign": "center",
                  }
                }
              >
                Press Cmd+R to reload,
Cmd+D or shake for dev menu
              </Text>
            </Text>
          </View>
        </View>
      </HomeScreen>
    </Connect(HomeScreen)>
  </Provider>
</App>
`;
